x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/tree/TreeTests.java

```diff
@@ -84,8 +87,14 @@ public static Tree buildRandomTree(int numFeatures, int depth) {
             }
             childNodes = nextNodes;
         }
+        List<String> categoryLabels = null;
+        if (randomBoolean()) {
+            categoryLabels = Arrays.asList(generateRandomStringArray(randomIntBetween(1, 10), randomIntBetween(1, 10), false, false));
+        }
 
-        return builder.build();
+        return builder.setTargetType(randomFrom(TargetType.values()))
+            .setClassificationLabels(categoryLabels)
```

Comment:
No, these types of "domain" validations are done in the `validate()` method. Not on `build()`.

I don't think we want to do validations every time we grab the config from the index. This has caused issues in the past and we have discussed moving our existing validations for Jobs, and datafeeds out of the builder and into the REST layer.

I think that these validations should be called at a level above the object so that they only need to be called on PUT. I